dl1_data_handler/image_mapper.py

```diff
@@ -1,744 +1,1112 @@
-import numpy as np
-import logging
-import bisect
+"""
+This module defines the ``ImageMapper`` classes, which holds the basic functionality for mapping raw 1D vectors into 2D mapped images.
+"""
 
+import numpy as np
 from scipy import spatial
 from scipy.sparse import csr_matrix
-from scipy.ndimage import rotate
-from astropy import units as u
-from collections import Counter
-
-logger = logging.getLogger(__name__)
+from collections import Counter, namedtuple
+
+from ctapipe.instrument.camera import PixelShape
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import Bool, Int
+
+__all__ = [
+    "ImageMapper",
+    "AxialMapper",
+    "BicubicMapper",
+    "BilinearMapper",
+    "NearestNeighborMapper",
+    "OversamplingMapper",
+    "RebinMapper",
+    "ShiftingMapper",
+    "SquareMapper",
+]
+
+# Constants for the ImageMapper classes
+Constants = namedtuple("Constants", ["decimal_precision", "tick_interval_limit"])
+constants = Constants(3, 0.002)
```

Comment:
Make it an ImageMapper class attribute (in this way, all objects will have the same value, but you won't have the global var.